R/coh_data_wrangling.R

```diff
@@ -498,25 +509,36 @@ get_immunization_dose <- function(data,
     vacc_date_col,
     min.len = 1L
   )
-  checkmate::assert_number(
-    immunization_delay,
-    lower = 0, finite = TRUE
+  stopifnot(
+    "Please provide a non-null integer number greater or equal than 0
+    in `outcome_delay`. Use round(`immunization_delay`,0)" =
+      checkmate::test_integerish(immunization_delay, lower = 0, null.ok = FALSE)
   )
   checkmate::assert_names(
     colnames(data),
     must.include = c(vacc_date_col, immunization_date_col)
   )
+  checkmate::assert_date(
+    data[[immunization_date_col]]
+  )
+  # The function can receive one or more column names in vacc_date_col
+  vacc_date_col_ <- vacc_date_col # hard coded to return right error message
```

Comment:
Is this extra assignment to a copy necessary? The loop below works fine as so:

```
for(i in vacc_date_col) { assert_date(data[[i]] }
```

The extra call to `c()` in `c(vacc_date_col_)` also seems unnecessary.

Reply:
I think it's necessary because when the iterator is `i`  the assertion error returns:
`Error: Assertion on 'data[[i]]' failed: Must be of class 'Date', not 'character'.`

This could be a bit confusing for the users (they won't know which one is the variable `i`). I would rather the message:
`Error: Assertion on 'data[[vacc_date_col]]' failed: Must be of class 'Date', not 'character'.`

Reply:
The extra call to `c()` is indeed unnecessary